File: generate-api-key.py
#!/usr/bin/env python3
"""
Generate a secure API key for Vector Storage Service
"""
import secrets
import os

def generate_api_key(length=32):
    """Generate a cryptographically secure API key"""
    return secrets.token_hex(length)

def create_env_file(api_key):
    """Create .env file with the API key"""
    env_content = f"""# Vector Storage Service Configuration
# Generated: {os.popen('date').read().strip()}

# Security - IMPORTANT!
API_KEY={api_key}

# CORS (Optional - comma separated origins)
# ALLOWED_ORIGINS=https://your-express-app.com,https://your-frontend.com

# Server Configuration (Optional)
# PORT=8000
# HOST=0.0.0.0
"""
    
    with open('.env', 'w') as f:
        f.write(env_content)
    
    print("✅ .env file created successfully!")
    print(f"📁 Location: {os.path.abspath('.env')}")

def main():
    print("🔐 Vector Storage Service - API Key Generator")
    print("=" * 50)
    
    # Check if .env already exists
    if os.path.exists('.env'):
        response = input("\n⚠️  .env file already exists. Overwrite? (y/N): ")
        if response.lower() != 'y':
            print("❌ Cancelled. Existing .env file kept.")
            return
    
    # Generate API key
    print("\n🔑 Generating secure API key...")
    api_key = generate_api_key()
    
    # Create .env file
    create_env_file(api_key)
    
    print("\n" + "=" * 50)
    print("🎉 Setup Complete!")
    print("=" * 50)
    print(f"\n📋 Your API Key:")
    print(f"   {api_key}")
    print("\n⚠️  Keep this key secret! Don't commit to Git.")
    print("\n📝 Next steps:")
    print("   1. Add .env to .gitignore (if not already)")
    print("   2. Set API_KEY on Railway/Cloud Run deployment")
    print("   3. Add API key to your Express server .env:")
    print(f"      VECTOR_STORAGE_API_KEY={api_key}")
    print("\n🚀 Ready to deploy!")
    print("\n📖 See INTEGRATION-GUIDE.md for usage examples")

if __name__ == "__main__":
    main()

File: generate-api-key.js
#!/usr/bin/env node
/**
 * Generate a secure API key for Vector Storage Service
 * Usage: node generate-api-key.js
 */

const crypto = require('crypto');
const fs = require('fs');
const path = require('path');
const readline = require('readline');

function generateApiKey(length = 32) {
  return crypto.randomBytes(length).toString('hex');
}

function createEnvFile(apiKey) {
  const envContent = `# Vector Storage Service Configuration
# Generated: ${new Date().toISOString()}

# Security - IMPORTANT!
API_KEY=${apiKey}

# CORS (Optional - comma separated origins)
# ALLOWED_ORIGINS=https://your-express-app.com,https://your-frontend.com

# Server Configuration (Optional)
# PORT=8000
# HOST=0.0.0.0
`;

  fs.writeFileSync('.env', envContent);
  console.log('✅ .env file created successfully!');
  console.log(`📁 Location: ${path.resolve('.env')}`);
}

async function promptUser(question) {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  return new Promise((resolve) => {
    rl.question(question, (answer) => {
      rl.close();
      resolve(answer);
    });
  });
}

async function main() {
  console.log('🔐 Vector Storage Service - API Key Generator');
  console.log('='.repeat(50));

  // Check if .env already exists
  if (fs.existsSync('.env')) {
    const response = await promptUser('\n⚠️  .env file already exists. Overwrite? (y/N): ');
    if (response.toLowerCase() !== 'y') {
      console.log('❌ Cancelled. Existing .env file kept.');
      return;
    }
  }

  // Generate API key
  console.log('\n🔑 Generating secure API key...');
  const apiKey = generateApiKey();

  // Create .env file
  createEnvFile(apiKey);

  console.log('\n' + '='.repeat(50));
  console.log('🎉 Setup Complete!');
  console.log('='.repeat(50));
  console.log('\n📋 Your API Key:');
  console.log(`   ${apiKey}`);
  console.log('\n⚠️  Keep this key secret! Don\'t commit to Git.');
  console.log('\n📝 Next steps:');
  console.log('   1. Add .env to .gitignore (if not already)');
  console.log('   2. Set API_KEY on Railway/Cloud Run deployment');
  console.log('   3. Add API key to your Express server .env:');
  console.log(`      VECTOR_STORAGE_API_KEY=${apiKey}`);
  console.log('\n🚀 Ready to deploy!');
  console.log('\n📖 See INTEGRATION-GUIDE.md for usage examples');
}

main().catch(console.error);
